src/config.js

```diff
@@ -307,6 +307,30 @@ export const SESSION_SECRET = fromEnvOrConfig("SESSION_SECRET", PRODUCTION ? und
 export const SESSION_MAX_AGE = fromEnvOrConfig("SESSION_MAX_AGE", 30 * 24 * 60 * 60); // 30d in seconds
 
 
+/**
+ * Flag indicating if Redis (via REDIS_URL) is required.
+ *
+ * Defaults to true if {@link PRODUCTION} and not {@link REVIEW_APP}, otherwise
+ * false.
+ *
+ * @type {boolean}
+ */
+/* XXX TODO: Provision lowest-tier Heroku Redis addon for review apps and
+ * remove the !REVIEW_APP condition.
+ *
+ * Heroku has good support for this, but to enable it I think we need to switch
+ * how we configure review apps.  Currently they're configured via the Heroku
+ * Dashboard¹.  I think we need to switch to the app.json file² instead, as
+ * described in the review app documentation.³
+ *   -trs, 12 Oct 2023
+ *
+ * ¹ <https://dashboard.heroku.com/pipelines/38f67fc7-d93c-40c6-a182-501da2f89d9d/settings>
+ * ² <https://devcenter.heroku.com/articles/app-json-schema>
+ * ³ <https://devcenter.heroku.com/articles/github-integration-review-apps>
+ */
+export const REDIS_REQUIRED = fromEnvOrConfig("REDIS_REQUIRED", PRODUCTION && !REVIEW_APP);
```

Comment:
Welp, this doesn't work at all when setting `REDIS_REQUIRED=false` in the environment or `"REDIS_REQUIRED": false` in the config because of the use of `||` in `fromEnvOrConfig()`. orz

I tested the functionality and it worked! but it only _appeared_ to work because I was testing against a previous version of the code which provided a default of `null` instead of `PRODUCTION && !REVIEW_APP`. In my testing, the production config set `REDIS_REQUIRED=true`, which worked, and the testing config set it to `false`, which worked, but the actual value was resolved to `null` (the default). This had the same effect as `false` and so masked the problematic use of `||`. When I changed the default, the behaviour changed, and I didn't re-test. Should have written proper tests, but I didn't because then we need an actual `REDIS_URL` for the connection. We could provide one in CI I suppose…

Reply:
https://github.com/nextstrain/nextstrain.org/pull/746